Userland/Utilities/shot.cpp

```diff
@@ -5,28 +5,100 @@
  */
 
 #include <AK/Format.h>
+#include <AK/Optional.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/DateTime.h>
 #include <LibCore/File.h>
 #include <LibGUI/Application.h>
 #include <LibGUI/Clipboard.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/Window.h>
 #include <LibGUI/WindowServerConnection.h>
 #include <LibGfx/PNGWriter.h>
 #include <unistd.h>
 
+class SelectableLayover final : public GUI::Widget {
+    C_OBJECT(SelectableLayover)
+public:
+    SelectableLayover(GUI::Window* window)
+        : m_window(window) {};
+    virtual ~SelectableLayover() override {};
+    Optional<Gfx::IntRect> region() const
+    {
+        return m_region;
+    }
+
+private:
+    virtual void mousedown_event(GUI::MouseEvent& event) override
+    {
+        if (event.button() != GUI::MouseButton::Left)
+            return;
+
+        if (m_achor_point.has_value())
+            m_achor_point.clear();
+        else
+            m_achor_point = event.position();
+    };
+
+    virtual void mousemove_event(GUI::MouseEvent& event) override
+    {
+        m_last_point = event.position();
+        if (m_achor_point.has_value())
+            update();
+    };
+
+    virtual void paint_event(GUI::PaintEvent&) override
+    {
+        m_region = build_rect(m_achor_point, m_last_point);
+        GUI::Painter painter(*this);
+        painter.draw_rect(m_region.value_or(Gfx::IntRect()), Gfx::Color::Black);
```

Comment:
The `build_rect` function basically already exists in `Rect<T>`, so you can do this instead of writing your own function. Also, don't instantiate a `GUI::Painter` to draw an zero-size rect.
```suggestion
        if (!m_achor_point.has_value() || !m_last_point.has_value()) {
            m_region = {};
            return;
        }
        GUI::Painter painter(*this);
        m_region = IntRect::from_two_points(*m_achor_point, *m_last_point);
```

Reply:
Thanks! Applied the changes.